libbeat/tests/compose/project.go

```diff
@@ -192,28 +195,72 @@ func (c *Project) KillOld(except []string) error {
 
 // Lock acquires the lock (300s) timeout
 // Normally it should only be seconds that the lock is used, but in some cases it can take longer.
+// Pid is written to the lock file, and it is used to check if process holding the process is still
+// alive to avoid deadlocks on unexpected finalizations.
 func (c *Project) Lock() {
 	timeout := time.Now().Add(300 * time.Second)
 	infoShown := false
 	for time.Now().Before(timeout) {
-		file, err := os.OpenFile(c.LockFile(), os.O_CREATE|os.O_EXCL, 0500)
-		file.Close()
-		if err != nil {
-			if !infoShown {
-				logp.Info("docker-compose.yml is locked, waiting")
-				infoShown = true
+		if acquireLock(c.LockFile()) {
+			if infoShown {
```

Comment:
not related to this PR, but I wonder if Debug would be better?

Reply:
After we added the conditions to print these messages only once I think that it is ok at the info level, so it informs that it couldn't continue because the lock is held, and then it informs when it can take it.
But it could be fine at the debug level too, yes.